terra_notebook_utils/utils.py

```diff
@@ -63,3 +66,22 @@ def is_notebook() -> bool:
         return "ZMQInteractiveShell" == get_ipython().__class__.__name__  # type: ignore
     except NameError:
         return False
+
+
+@lru_cache()
+def get_execution_context() -> ExecutionContext:
+    """
+    Identify information about the context in which terra-notebook-utils is executing.
+    Currently, this determination is made based on the presence and value of the WORKSPACE_BUCKET.
+    Other/improved means may be identified in the future.
+    """
+    execution_environment = ExecutionEnvironment.OTHER
+    execution_platform = ExecutionPlatform.UNKNOWN
+    workspace_bucket = os.environ.get('WORKSPACE_BUCKET', None)
+    if workspace_bucket:
+        execution_environment = ExecutionEnvironment.TERRA_WORKSPACE
+        if workspace_bucket.startswith("gs://"):
+            execution_platform = ExecutionPlatform.GOOGLE
+        elif workspace_bucket.startswith("https://"):
```

Comment:
I don't think we can depend on just using https.  Maybe we could do something like
```suggestion
        elif re.search("^https://.*\.(dfs|blob)\.core\.windows\.net$", workspace_bucket):
```

(assuming you have `import re` above)

Reply:
Based on recent Slack messages, it sounds like the env vars to be used for Terra Azure Cloud Environments, and what they represent, is in flux.
We can revisit this when that settles out.
And, yes, happy to use a more specific regex pattern match if we have a pattern that we are confident will always be correct.

Reply:
Ah ok gotcha

Reply:
Opened this ticket and noted the above, to be addressed in the (near) future:  
[Reliable identification of primary cloud #403](https://github.com/DataBiosphere/terra-notebook-utils/issues/403)